cache/cache.go

```diff
@@ -20,20 +21,37 @@ type Cache struct {
 	cacheFile   string
 	allDirty    bool
 	dirtyFiles  []string
+
+	opts []shell.RunOption
 }
 
-func New() *Cache {
+func New(opts ...shell.RunOption) *Cache {
 	return &Cache{
-		ranOnce:   make(map[*taskrunner.Task]bool),
-		cacheFile: path.Join(CacheDir, "example.json"),
+		ranOnce: make(map[*taskrunner.Task]bool),
+		opts:    opts,
```

Comment:
could you add an example usage in the commit message?

Reply:
sg